### ISSUE TITLE
Cannot write to closing transport

### ISSUE THREAD

**Issue by KristianOellegaard** [ReneNulschDE/mbapi2020#148]
When trying to set Max SoC I some times get the following in my automation. Restart solves the problem.

Stopped because an error was encountered at November 6, 2023 at 8:37:05 PM (runtime: 0.10 seconds)

Cannot write to closing transport

Any idea what could be wrong?

**Comment by Joo01** [comment]
Same issue here!

```
Logger: homeassistant
Source: custom_components/mbapi2020/websocket.py:164
Integration: MercedesME 2020 (documentation, issues)
First occurred: 7. November 2023 um 10:42:24 (1 occurrences)
Last logged: 7. November 2023 um 10:42:24

Error doing job: Task exception was never retrieved
Traceback (most recent call last):
  File "/config/custom_components/mbapi2020/websocket.py", line 164, in _recv
    data = await self._connection.receive()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/aiohttp/client_ws.py", line 265, in receive
    await self.pong(msg.data)
  File "/usr/local/lib/python3.11/site-packages/aiohttp/client_ws.py", line 146, in pong
    await self._writer.pong(message)
  File "/usr/local/lib/python3.11/site-packages/aiohttp/http_websocket.py", line 670, in pong
    await self._send_frame(message, WSMsgType.PONG)
  File "/usr/local/lib/python3.11/site-packages/aiohttp/http_websocket.py", line 646, in _send_frame
    self._write(header + mask + message)
  File "/usr/local/lib/python3.11/site-packages/aiohttp/http_websocket.py", line 663, in _write
    raise ConnectionResetError("Cannot write to closing transport")
ConnectionResetError: Cannot write to closing transport
```

**Comment by ReneNulschDE** [comment]
Could you check the logfile if there are other MB2020 messages (Info, Warning, Error) near to the error message? esp line like these:


```
INFO (MainThread) [custom_components.mbapi2020.oauth] Start async_refresh_access_token() with refresh_token
INFO (MainThread) [custom_components.mbapi2020.websocket] Connecting to wss://websocket.emea-prod.mobilesdk.mercedes-benz.com/ws
INFO (MainThread) [custom_components.mbapi2020.websocket] Connected to mercedes websocket at wss://websocket.emea-prod.mobilesdk.mercedes-benz.com/ws

```

**Comment by ReneNulschDE** [comment]
@KristianOellegaard : Could you share the stack trace please?
@Joo01 : The error appears in the ping/pong operation to keep the tunnel open. The pong cant be send when the tunnel gets recreated. Is any operation affected besides the unneeded error message?

**Comment by ReneNulschDE** [comment]
I have [added](https://github.com/ReneNulschDE/mbapi2020/commit/1a8e5d2fd8fd6d38bd3000ffffb518f954d8ff52) some additional error handling for this type of error. I'll test it a few days in my environment and will publish a new version next week.

**Comment by ReneNulschDE** [comment]
Solved with v0.9.6
